### ISSUE TITLE
VecElement misbehaves with Bools, especially when using varargs.

### ISSUE THREAD

**Issue by Maaarcocr** [JuliaLang/julia#30056]
As an example this code:
```
function f(args...)
    println(args)
end

vectorize(x) = VecElement(x)
f(vectorize.((true,true,true,true)), vectorize.((1,2,3,4))) 
```
produces this output: 
```
((VecElement{Bool}(false), VecElement{Bool}(false), VecElement{Bool}(false), VecElement{Bool}(false)), (VecElement{Int64}(1), VecElement{Int64}(2), VecElement{Int64}(3), VecElement{Int64}(4)))�
```

I've also seen `VecElement{Bool}`s changing value in other circumstances, but this example is the simplest I could reproduce in isolation.

**Comment by chethega** [comment]
```
julia> a,b = vectorize.((true,true,true,true)), vectorize.((1,2,3,4))
((VecElement{Bool}(true), VecElement{Bool}(true), VecElement{Bool}(true), VecElement{Bool}(true)), (VecElement{Int64}(1), VecElement{Int64}(2), VecElement{Int64}(3), VecElement{Int64}(4)))

julia> @code_llvm f(a,b)

; Function f
; Location: REPL[11]:2
; Function Attrs: sspstrong
define void @julia_f_953096378(<4 x i8>, <4 x i64>) #0 {
top:
  %args = alloca { <4 x i8>, <4 x i64> }, align 32
  %.sroa.0.0..sroa_cast = bitcast { <4 x i8>, <4 x i64> }* %args to <4 x i64>*
  store <4 x i64> %1, <4 x i64>* %.sroa.0.0..sroa_cast, align 32
  %2 = addrspacecast { <4 x i8>, <4 x i64> }* %args to { <4 x i8>, <4 x i64> } addrspace(11)*
  call void @julia_println_953096193({ <4 x i8>, <4 x i64> } addrspace(11)* nocapture readonly %2)
  ret void
}
```
So here we have the problem. The emitted llvm code looks not just logically wrong, but wrong on its own merits (regardless of the source code for `f`).

**Comment by KristofferC** [comment]
Crashes in 1.3 and fixed in master (will be in 1.4) by (likely) https://github.com/JuliaLang/julia/pull/34473.
